addons/io_scene_gltf2/io/exp/gltf2_io_user_extensions.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2019 The glTF-Blender-IO authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def export_user_extensions(hook_name, export_settings, gltf2_object, *args):
+    if gltf2_object.extensions is None:
+        gltf2_object.extensions = {}
```

Comment:
I think this should be removed (user-extensions can do it themselves) and `gltf2_object` should become part of `args`. Since (a) not all user-extensions are going to do anything to the extensions property, and (b) it makes hooks more flexible, since their first argument doesn't have to be an object with an extensions property.

Reply:
Sure the user can do that himshelf, but since I have focused on user extensions, I think it was good to just provide the empty dictionary so he dont need to do the check. 

I think I dont get the second part. How does it makes the hooks more flexible? If I add the gltf node to the args, the user still needs to receive those args and parse them.

Reply:
It allows hooks that do not pass a glTF object as their first argument. 

I was going to say if you want to have it here, you can delete the check from the example user-extension, but I think robustness demands it be there anyway, since another user-extension may have set it to `None`.

Reply:
Do we want to support hooks that doesnt pass a gltf object? What would they do?

Reply:
Modify the export settings before exporting begins.

Modify an internal primitive (which is a dict).

Modify the "schedule" for making animations from a collection of object/action pairs.

Anyway it seems better to preserve flexibility rather than try to predict what all possible desirable hooks will look like.